### DISCUSSION TITLE
Alarm control panel Issue only showing state on Home Assistant

### DISCUSSION THREAD

**Discussion by tjtjdgur2719** [taligentx/dscKeybusInterface#140] [dscKeybusInterface Discussion]
I managed to setup everything using wemos d1 mini with home assistant.
Everything works perfectly with all the binary sensors showing up and running.
However, alarm_control_panel (Security System Partition 1) on home assistant does show up and is not unavailable state and if i use physical dsc keypad in the house it shows up the changing state but if i try to click arm on home assistant it does nothing .. the code is correct and matching dsc password. 
Any help?? Thanks

**Comment by TrendMend** [comment]
Did you ever figure this out? Thank you.

**Comment by tjtjdgur2719** [comment]
Sadly no..plus regular sensor state gets frozen after few days have to restart every time

**Comment by taligentx** [comment]
Hi @tjtjdgur2719,

> (..) but if i try to click arm on home assistant it does nothing 

Does sending keys work with the `KeybusReader` sketch?  You should be able to type in a key via serial terminal, have the panel recognize the keypress, and see the key output in the keybus data - this verifies that the wiring for writing is functional.

> .plus regular sensor state gets frozen after few days have to restart every time

For esp8266/esp32, try the current develop branch - there was a bug that could cause crashes.  If on the current develop branch, does the esp8266 stop publishing updates to MQTT?  Can you still ping it after it becomes unresponsive in HA?

**Comment by TrendMend** [comment]
Okay so funny thing, I was actually able to make it work by sending
commands via mqtt explorer, and swapping the arm disarm codes in home
assistant. By doing so when I pressed disarm it armed and caught the state.
Then I was able to swap them back and it is functioning now.

**Comment by taligentx** [comment]
HI @TrendMend - good to hear it was resolved, odd that the states were out of sync but feel free to post an issue if that happens again.

@tjtjdgur2719 - another possibility specific to the Wemos D1 - some of the clones have low current voltage regulators and can cause reboots/crashes:

https://www.letscontrolit.com/forum/viewtopic.php?t=6603

You can try using something like one of the LM2596 voltage regulator modules set to 3.3v and connect directly to the Wemos 3.3v pin to bypass the onboard regulator.  I've updated the develop branch README.md to note this as a common issue.

**Comment by tjtjdgur2719** [comment]
Wow Thank you @taligentx @TrendMend for the detail info & support!

I will try out all the suggestions and see if it works. 

Im currently out in town for a while so I will post an update when I get to work on it.

Thanks!

**Comment by tjtjdgur2719** [comment]
So update on this I swapped to LM2596 voltage regulator module to 5V pin on Wemos d1 mini and so far no crashes with updated sketch. No freezing sensors. 
However, the buttons for arm/disarm on home assistant is still not working. I found out dsc/set is not working properly. how is it suppose to be set?
I have tried changing configuration script to see how things work..

alarm_control_panel:
 - platform: mqtt
    name: "Security System Panel"
    state_topic: "dsc/Get/Partition1"
    availability_topic: "dsc/Status"
    command_topic: "dsc/Set"
    payload_disarm: "1D"
    payload_arm_home: "1S"
    payload_arm_away: "1A"
    payload_arm_night: "1N"

to: 

alarm_control_panel:

  - platform: mqtt
    name: "Security System Panel"
    state_topic: "dsc/Get/Partition1"
    command_topic: "dsc/Get/Partition1"
    availability_topic: "dsc/Status"
    payload_disarm: "disarmed"
    payload_arm_home: "armed_home"
    payload_arm_away: "armed_away"
    payload_arm_night: "armed_night"

I was only able to change mqtt state but not actually apply to the dsc alarm system. 
Im guessing what I did was just trying to change received status to be seen as new status.

How can I make the command_topic: "dsc/Set" work? 

Thank you in advance 

![Original Configuration](https://user-images.githubusercontent.com/29513388/108421133-b9583980-7202-11eb-9716-173dc107c352.png)

**Reply by kricon** [reply]
Have you tried flashing `KeybusReader` and verified that typing in keys in serial monitor is being recognized from panel, as said above? Even if you don't need virtual keypad, wiring (with transistor) for it is required to send any dsc/set commands to the panel (such as arming/disarming thru HomeAssistant). Without it, using HomeAssistant would be read-only from the panel.
